docs/standard/base-types/regular-expression-options.md

```diff
@@ -144,9 +144,11 @@ The following example modifies the regular expression pattern from the previous
 
 ## Multiline mode
 
-The <xref:System.Text.RegularExpressions.RegexOptions.Multiline?displayProperty=nameWithType> option, or the `m` inline option, enables the regular expression engine to handle an input string that consists of multiple lines. It changes the interpretation of the `^` and `$` language elements so that they match the beginning and end of a line, instead of the beginning and end of the input string.
+The <xref:System.Text.RegularExpressions.RegexOptions.Multiline?displayProperty=nameWithType> option, or the `m` inline option, enables the regular expression engine to handle an input string that consists of multiple lines. It changes the interpretation of the `^` and `$` language elements so that they indicate the beginning and end of a line, instead of the beginning and end of the input string.
 
-By default, `$` matches only the end of the input string. If you specify the <xref:System.Text.RegularExpressions.RegexOptions.Multiline?displayProperty=nameWithType> option, it matches either the newline character (`\n`) or the end of the input string. It does not, however, match the carriage return/line feed character combination. To successfully match them, use the subexpression `\r?$` instead of just `$`.
+By default, `$` will be satisfied only at the end of the input string. If you specify the <xref:System.Text.RegularExpressions.RegexOptions.Multiline?displayProperty=nameWithType> option, it will be satisfied by either the newline character (`\n`) or the end of the input string.
+
+In neither case does `$` recognize the carriage return/line feed character combination (`\r\n`). `$` always ignores any carriage return (`\r`). To end your match with either `\r\n` or `\n`, use the subexpression `\r?$` instead of just `$`. Note that this will make the `\r` part of the match - if you are accessing the match, and do not want `\r` to be part of it, you may replace `\r?$` with a positive lookahead assert, like this `(?=\r\n|\z|(?<!\r)(?=\n))`. In this expression, `\z` asserts the end of the input string. To do the same without multiline mode, where `$` indicates the end of the input string, use `(?=\r\n\z|\z|(?<!\r)(?=\n\z))`.
```

Comment:
explanation (detail in https://github.com/dotnet/runtime/issues/25598#issuecomment-666955410) -- in our engine, $ is satisfied twice in `xxxxx\n` (by both the `\n` and the end) so it should be satisfied twice in `xxxx\r\n` (by both the `\r\n` and the end)

so for multiline mode -- 
`(?=\r\n|\z|(?<!\r)(?=\n))` will be satisfied by `\r\n`, `\z` (end), and `\n` (so long as it's not directly after `\r`)

for non multiline mode -- 
`(?=\r\n\z|\z|(?<!\r)(?=\n\z))` is the same, but any `\n` must be at the end.